src/components/Fields/TextField.js

```diff
@@ -260,11 +260,12 @@ class TextField extends Component {
     }
 
     render() {
+        const hasValue = this.state.value === null ? false : `${this.state.value}`;
```

Comment:
How about this instead:
`const hasValue = this.state.value === 0 ? true : Boolean(val)`
?

That way `hasValue` will always be a boolean like its name suggests instead of sometimes being a string.

```
Boolean(undefined) => false
Boolean(null) => false
Boolean('') => false
Boolean(' ') => true
Boolean('test') => true
Boolean('0') => true
```